tests/integration/devfile/cmd_devfile_registry_test.go

```diff
@@ -81,41 +79,11 @@ var _ = Describe("odo devfile registry command tests", func() {
 	})
 
 	When("using a git based registries", func() {
-		var deprecated, docLink, out, err, co string
-		BeforeEach(func() {
-			deprecated = "Deprecated"
-			docLink = "https://github.com/redhat-developer/odo/tree/main/docs/public/git-registry-deprecation.adoc"
-		})
-
-		It("should not show deprication warning, if git registry is not used", func() {
-			out, err = helper.Cmd("odo", "preference", "registry", "list").ShouldPass().OutAndErr()
-			helper.DontMatchAllInOutput(fmt.Sprintln(out, err), []string{deprecated, docLink})
-		})
-
+		var err string
 		When("adding git based registries", func() {
-			BeforeEach(func() {
-				out, err = helper.Cmd("odo", "preference", "registry", "add", "RegistryFromGitHub", "https://github.com/devfile/registry").ShouldPass().OutAndErr()
-
-			})
-			It("should show deprecation warning", func() {
-				co = fmt.Sprintln(out, err)
-				helper.MatchAllInOutput(co, []string{deprecated, docLink})
-
-				By("odo preference registry list is executed, should show the warning", func() {
-					out, err = helper.Cmd("odo", "preference", "registry", "list").ShouldPass().OutAndErr()
-					co = fmt.Sprintln(out, err)
-					helper.MatchAllInOutput(co, []string{deprecated, docLink})
-				})
-				// TODO: Should `odo init` support GitHub based registries
-				// By("should successfully delete registry", func() {
-				// 	out, err = helper.Cmd("odo", "init", "--name", "aname", "--devfile", "nodejs", "--devfile-registry", "RegistryFromGitHub").ShouldPass().OutAndErr()
-				// 	co = fmt.Sprintln(out, err)
-				// 	helper.MatchAllInOutput(co, []string{deprecated, docLink})
-				// })
-			})
-			It("should not show deprecation warning if git registry is not used for component creation", func() {
-				out, err = helper.Cmd("odo", "init", "--name", "aname", "--devfile", "nodejs", "--devfile-registry", "DefaultDevfileRegistry").ShouldPass().OutAndErr()
-				helper.DontMatchAllInOutput(fmt.Sprintln(out, err), []string{deprecated, docLink})
+			It("should fail with error", func() {
```

Comment:
You have two When's without any BeforeEach, you can remove both of them and have only one `It("should fail adding a git based registry", ...`